### PR TITLE
Reorganized package, put example it it's own directory, etc

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
@natebosch @nshahan – this is a good start, I think – PTAL

File: webdev/lib/src/command/build_runner_command_base.dart [hunk 1]
```diff
@@ -19,13 +24,66 @@ abstract class BuildRunnerCommandBase extends Command {
...
-    final executable = 'pub';
-    final arguments = ['run', 'build_runner', 'generate-build-script'];
-    final results = await Process.run(executable, arguments);
-    return new Uri.file(results.stdout.toString().trim());
+  Future runCore(String command) async {
+    final arguments = [command]..addAll(argResults.arguments);
+    var exitPort = new ReceivePort();
+    await Isolate.spawnUri(await _buildRunnerScript(), arguments, null,
+        onExit: exitPort.sendPort, automaticPackageResolution: true);
+    await exitPort.first;
```
Comment (natebosch) [inline webdev/lib/src/command/build_runner_command_base.dart]: As an FYI this isn't guaranteed to fire.
See all the extra handling we have for exception cases: https://github.com/dart-lang/build/blob/0c77443dd74edda706e89189c5ccfb70b06a22d1/build_runner/bin/build_runner.dart#L58-L85

Comment (kevmoo) [reply webdev/lib/src/command/build_runner_command_base.dart]: There oughta be a helper...

Comment (kevmoo) [reply webdev/lib/src/command/build_runner_command_base.dart]: Can be a todo? File an issue?

Comment (natebosch) [reply webdev/lib/src/command/build_runner_command_base.dart]: yes there should :)

I still haven't made up my mind whether I want a specific helper in `build_runner` or whether this we should make a more user friendly API around isolates in package:io

**Comment by kevmoo** [comment]
Or https://pub.dartlang.org/packages/isolate :-)

File: webdev/pubspec.yaml [hunk 2]
```diff
@@ -2,17 +2,15 @@ name: webdev
...
-      url: https://github.com/dart-lang/build.git
-      path: build_runner
-  build_web_compilers: ^0.1.1
+  io: ^0.3.2+1
+  stack_trace: ^1.9.2
 
-dev_dependencies:
-  test: "^0.12.0"
+executables:
+  webdev:
```
Comment (natebosch) [inline webdev/pubspec.yaml]: just to confirm - we don't plan on publishing this until we've implemented the upper bound version check on build_runner right?

**Comment by kevmoo** [comment]
Yup!

File: webdev/pubspec.yaml (hunk 2 above)
Comment (nshahan) [reply webdev/pubspec.yaml]: I agree, not ready to publish until we have locked down the validation parts to make sure people don't get a rough introduction.

File: example/pubspec.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+name: webdev_example_app
+description: A web app example for webdev CLI.
+
+environment:
+  sdk: ">=2.0.0-dev.32.0 <2.0.0"
+
+dependencies:
+  angular: ^5.0.0-alpha+3
```
Comment (nshahan) [inline example/pubspec.yaml]: Can we bump to alpha+8?

File: webdev/pubspec.yaml (hunk 2 above)
Comment (kevmoo) [reply webdev/pubspec.yaml]: ack

File: example/pubspec.yaml (hunk 3 above)
Comment (kevmoo) [reply example/pubspec.yaml]: uh...sure

**Pull request merged**